api_views/books.py

```diff
@@ -77,4 +77,47 @@ def get_by_title(book_title):
                 }
                 return Response(json.dumps(responseObject), 200, mimetype="application/json")
             else:
-                return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
\ No newline at end of file
+                return Response(error_message_helper("Book not found!"), 404, mimetype="application/json")
+
+
+def get_by_title_v2(book_title):
+    hardcoded_resp = "user123"  # Hardcoded username for testing purposes
+
+    if hardcoded_resp == "user123":
+        if vuln:
+            try:
+                book = Book.query.filter_by(book_title=book_title).first()  # Removed type conversion to str
+                if book:
+                    return Response(
+                        json.dumps({
+                            'book_title': book.book_title,
+                            'secret': book.secret_content,
+                            'owner': book.user.username
+                        }),
+                        200, mimetype="application/json"
+                    )
+                else:
+
+                    return Response("Error occurred!", 404, mimetype="application/json")
+            except Exception as e:
```

Comment:
Generic exception handling could mask specific errors. Error response should include more details about the exception